### PR TITLE
[FIX] Fixes the unpack/pack operations for bgzf streams.

### PR DESCRIPTION
**Pull request by rrahn**

### CODE REVIEW

File: include/seqan3/contrib/stream/bgzf_stream_util.hpp [hunk 1]
```diff
@@ -139,13 +140,17 @@ compressInit(CompressionContext<detail::bgzf_compression> & ctx)
 inline uint16_t
 _bgzfUnpack16(char const * buffer)
 {
-    return detail::to_little_endian(*reinterpret_cast<uint16_t const *>(buffer));
+    uint16_t tmp;
+    std::uninitialized_copy(buffer, buffer + sizeof(uint16_t), reinterpret_cast<char *>(&tmp));
```
Comment (smehringer) [inline include/seqan3/contrib/stream/bgzf_stream_util.hpp]: Why not memcpy?

**Comment by h-2** [comment]
Why are these changes necessary?

**Comment by marehr** [comment]
> Why are these changes necessary?

Because of valid sanitizer issues.

The problem is that 
```c++
*reinterpret_cast<uint16_t const *>(buffer)
```

of a

```c++
char const * buffer
```
is unsafe, because uint16_t has a different data alignment then char. So the reinterpret_cast is unsafe / undefined behaviour.

The fix first copies the corresponding bytes into a proper memory region and then access it.

**Comment by h-2** [comment]
Ah, I didn't think about alignment issues.

**Pull request merged**